### PR TITLE
RavenDB-14948 Log retention time modified to check last write time to let hours resolution

### PR DESCRIPTION
**Pull request by haludi**
https://issues.hibernatingrhinos.com/issue/RavenDB-14948

### CODE REVIEW

**Comment by haludi** [comment]
For hours resolution we can't use the name of the file (unless we are adding the hour to the name)
So I used the last write time to apply the retention policy.
So if something will modify the log it will "reset" the retention timer for this file.
Maybe this is even a benefit.

File: src/Sparrow/Logging/LoggingSource.cs [hunk 1]
```diff
@@ -363,25 +363,9 @@ public LogInfo(string fileName)
...
-                if (start < 0)
-                {
-                    dateTime = DateTime.MinValue;
-                    return false;
-                }
-
-                var date = fileName.Substring(start, DateFormatLength);
-                return DateTime.TryParse(date, out dateTime);
+                var fileInfo = new FileInfo(fileName);
+                dateTime = fileInfo.LastWriteTime;
```
Comment (ayende) [inline src/Sparrow/Logging/LoggingSource.cs]: Can this throw if the value does not exists?

**Review by ayende — APPROVED** [review]

File: src/Sparrow/Logging/LoggingSource.cs (hunk 1 above)
Comment (grisha-kotler) [inline src/Sparrow/Logging/LoggingSource.cs]: This will throw an exception if the file doesn't exist

Comment (grisha-kotler) [reply src/Sparrow/Logging/LoggingSource.cs]: nevermind, this will throw for a directory

**Review by grisha-kotler — APPROVED** [review]

**Pull request merged**